server.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+const Client = require('./model/client.js');
+const PORT = process.env.PORT || 3000;
+const server = net.createServer();
+const ee = new EE();
+
+const pool = [];
+
+ee.on('@dm', (client, string) => {
+  let nickname = string.split(' ').shift().trim();
+  let message = string.split(' ').splice(1).trim();
+
+  pool.forEach((client) => {
+    if(client.nickname === nickname)
+      client.socket.write(`${client.nickname}: ${message}`);
+  });
+});
+
+ee.on('@all', (client, string) => {
+  pool.forEach((c) => {
+    c.socket.write(`${client.nickname}: ${string}`);
+  });
+});
+
+ee.on('@rename', (client, string) => {
+  client.nickname = string.split(' ').shift().trim();
+});
+
+ee.on('default', (client, string) => {
+  client.socket.write('not a command \n');
+});
+
+server.on('connection', (socket) => {
+  var client = new Client(data);
```

Comment:
The sockets that I meant needed to be changed here is the one in the socket .on('data', (socket) => { below. Sorry if I messed up this comment. The cient(socket) to the client I believe is fine but when you call data.toString() on 41 it broke.